packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -302,8 +302,17 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
     if (hasTwoColumnItems) {
       // If the number of items to position is greater that the batch size
       // we identify the batch with the two column item and apply the graph only to those items
-      const splitIndex = calculateSplitIndex(itemsWithoutPositions);
-      const shouldBatchItems = itemsWithoutPositions.length > TWO_COL_ITEMS_MEASURE_BATCH_SIZE;
+      const twoColumnIndex = itemsWithoutPositions.indexOf(twoColumnItems[0]);
+
+      const skipGraph =
+        heights.every((height) => height === 0) && twoColumnIndex < heights.length - 1;
+      const shouldBatchItems =
```

Comment:
if skipGraph is true do we still need to batch items? or is this just to keep the code simple since it doesn't really matter anyways

Reply:
Exactly, its not really needed but we would execute the same steps as when there is a batch, I did it like this so we are not repeating code but that is another option on how to handle this.

Reply:
sg, can you leave a comment here around this?